tests/integration/profile_test.py

```diff
@@ -119,6 +145,21 @@ def dummy_inactive_profile():
         cmdlib.exec_cmd(_nmcli_delete_connection("testProfile"))
 
 
+@contextmanager
+def create_active_profile(con_name):
```

Comment:
It does much more than the name implies and it is hard to understand what is the test doing unless diving in this implementation.
How about this as a fixture:
```
cloned_profile_name = "test-profile
with cloned_active_profile(name=cloned_profile_name, source=con_name):
    assert _profile_exists(ETH_PROFILE_DIRECTORY + "ifcfg-" + cloned_profile_name)
    cmdlib.exec_cmd(_nmcli_delete_connection(con_name))
    yield cloned_profile_name
```

Reply:
I changed the name to "cloned_active_profile". I found hard to understand using it as a fixture. Please, look at the new approach I think it is clearer now :-)